### DISCUSSION TITLE
Looking for an exemple of Curl Call to Identify

### DISCUSSION THREAD

**Discussion by boivinx7** [stashapp/stash#5071] [Q&A] [answered]
So i've been looking here and there at the API with the /playground but i cant find anything.
I'm just looking for an exemple to do Curl calls if the API with Identify with X meta provider and X path.
i would like to automate those in Crontab, i already have scan and generate. 

Thanks for anyone wanting to assist, will still continue to try to find on my own, but hey you never know.

**Comment by Maista6969** [comment]
There are examples for this in [the documentation](https://docs.stashapp.cc/networking/api/)

**Comment by boivinx7** [comment]
> There are examples for this in [the documentation](https://docs.stashapp.cc/networking/api/)

Yes i know, but not for what im looking for. 
trying to get Identify to work, i already have Scan and Generate working

**Comment by Maista6969** [comment]
The documentation explorer in the playground has a nice search feature 🙂

Looking up "identify" you can find the mutation you need as well as the schema for the input it requires
![bilde](https://github.com/user-attachments/assets/b572d42e-3fda-4b16-a44d-321ba0357546)

So you'll probably want something like this
```gql
mutation {
  metadataIdentify(
    input: {paths: ["/path/to/files"], sources: [{source: {stash_box_endpoint: "https://stashdb.org/graphql"}}]}
  )
}
```

**Comment by boivinx7** [comment]
Great thanks, I just seem to not be good at understanding that API stuff, Ive done Tones of PS modules, with plex, Servarr, Teamviewer API's but that one i dont seem to get it.

Anyway someone can tell me how to get listScrapers to work ?
After that i would be all set i think


`query {  
  listScrapers
  {
    id
    name
  }
}`

**Comment by boivinx7** [comment]
Scratch that found it

`query {  
  listScrapers (types : SCENE)
  {
  	id
    name
  }
}`

**Comment by boivinx7** [comment, marked as answer]
if anyone would know how to format it for it to work for a Curl Call

`'{ "query": "mutation { metadataIdentify ( input : { paths: ["/path/to/files"] sources: [{source: {stash_box_endpoint: "https://stashdb.org/graphql"}}] } ) }" }'`

Trying to check it to convert to and from Json nothing really seems to work

**Comment by boivinx7** [comment]
Ive asked CoPilot, and got it working all good now

**Reply by wiffleman** [reply]
curious - what paths did you map in the /path/to/files?

tryign to get this to work but can't seem to get the curl command to run...

**Reply by boivinx7** [reply]
> curious - what paths did you map in the /path/to/files?
> 
> tryign to get this to work but can't seem to get the curl command to run...

Its the path of the file like you would select in the UI

**Reply by wiffleman** [reply]
thanks. i still can't quite figure this out. would you mind sharing your entire curl command? 

for context i am trying to run this as a scheduled task on DSM (Synology)

Example code
curl -H "ApiKey: APIKEYHERE" -H "Content-Type: application/json" --data '{ "query": "mutation { metadataIdentify ( input : { paths: ["/path/to/files"] sources: [{source: {stash_box_endpoint: "https://stashdb.org/graphql"}}] } ) }" }' localhost:9999/graphql
